.circleci/config.yml

```diff
@@ -0,0 +1,161 @@
+      - checkout
+      - save_cache:
+          key: v1-repo-{{ .Environment.CIRCLE_SHA1 }}
+          paths:
+            - ~/buildhub
+
+  build_test_and_deploy:
+    docker:
+      - image: ubuntu:16.04
+    working_directory: ~/buildhub
+    steps:
+      - run:
+          name: Install essential packages
+          command: |
+            apt-get update && apt-get install -y ca-certificates curl python3-pip zip
+
+      - restore_cache:
+          keys:
+            - v1-repo-{{ .Environment.CIRCLE_SHA1 }}
+
+      - run:
+          name: Install Docker
+          command: |
+            set -x
+            VER="17.09.0-ce"
+            curl -L -o /tmp/docker-$VER.tgz https://download.docker.com/linux/static/stable/x86_64/docker-$VER.tgz
+            tar -xz -C /tmp -f /tmp/docker-$VER.tgz
+            mv /tmp/docker/* /usr/bin
+
+      - run:
+          name: Install Docker Compose
+          command: |
+            set -x
+            curl -L https://github.com/docker/compose/releases/download/1.16.0/docker-compose-`uname -s`-`uname -m` > /usr/local/bin/docker-compose
+            chmod +x /usr/local/bin/docker-compose
+
+      - setup_remote_docker:
+          version: 17.09.0-ce
+
+      - run:
+          name: Create version.json
+          command: |
+            # create a version.json per https://github.com/mozilla-services/Dockerflow/blob/master/docs/version_object.md
+            printf '{"commit":"%s","version":"%s","source":"https://github.com/%s/%s","build":"%s"}\n' \
+            "$CIRCLE_SHA1" \
+            "$CIRCLE_TAG" \
+            "$CIRCLE_PROJECT_USERNAME" \
+            "$CIRCLE_PROJECT_REPONAME" \
+            "$CIRCLE_BUILD_URL" > version.json
+
+      - run:
+          name: Start kinto
+          command: |
+            # This assures that the `docker-compose up`, later, isn't
+            # put to do the build in the background if it needs to be built.
+            # Basically, by being explicit about it here and now we can
+            # we certain the image is ready by the time we start it ("up")
+            # in the background ("-d").
+            docker-compose -f docker-compose.ci.yml build kinto
+
+            # Use -d to put it in the background "docker style".
+            docker-compose -f docker-compose.ci.yml up -d kinto
+
+      - run:
+          name: Run functional tests
+          command: |
+            docker-compose -f docker-compose.ci.yml run kinto initialize-kinto-wizard jobs/buildhub/initialization.yml  --server http://kinto:8888/v1 --auth user:pass
+            docker-compose -f docker-compose.ci.yml run buildhub functional-tests
+
+      - run:
+          name: Lint check
+          command: |
+            docker-compose -f docker-compose.ci.yml run buildhub lintcheck
+
+      - run:
+          name: Mock run building lambda.zip.
+          command: |
+            docker-compose -f docker-compose.ci.yml run buildhub lambda.zip
+
+      - run:
```

Comment:
@bqbn Hey,
These below lines I took form [a working circle 2.0 project](https://github.com/mozilla-services/tecken/blob/master/.circleci/config.yml) that @milescrabill helped me build. 

Basically I just copied it. And replaced the name `tecken` with `buildhub`. 
See [the script too](https://github.com/peterbe/buildhub/blob/65910b885b47dd6bc4b5a873b2288a7cd0fb521c/bin/deploy-dockerhub.sh)

The build image name is specified in the [`docker-compose.ci.yml`](https://github.com/peterbe/buildhub/blob/65910b885b47dd6bc4b5a873b2288a7cd0fb521c/docker-compose.ci.yml#L21) as `mozilla/buildhub`. 

I appreciate that it's a bit big to review but if you could please take look at the lines specifically for the "Push to Dockerhub". 

I have no idea how to test any of this. And if we don't get it right here in this monster PR, we can admittedly iterate over it later.

Reply:
The below looks good to me. The way I would test it is just let CircleCI build it and see if what errors come out of it. If success, then the log should show it successfully pushed an image to dockerhub.